### PR TITLE
Fix: Assign exact end values to Node properties when animations finish

### LINKED ISSUE #266: Issue with handling floating-point precision in node property animation
It seems to be an issue with handling floating-point values during node property animation. When animating a property in a node to a target float value, it's observed that instead of setting the exact target value upon animation completion, the node's property ends up with a nearby floating-point value that may have additional decimal places.

Observed this behaviour with most of the node proeprties. 

For example, if the initial value of an alpha property on a node is 1, and I animate it to 0.1, after the animation completes, the alpha value on the node might not be exactly 0.1, but rather something like 0.09999999991. 

This issue leads to some other issue in Blits transition

### PR DESCRIPTION
**Pull request by marcel-danilewicz-consult-red**
Add setting node properties to the end values after the animation is finished

### CODE REVIEW

File: src/core/animations/CoreAnimation.ts [hunk 1]
```diff
@@ -198,6 +198,11 @@ export class CoreAnimation extends EventEmitter {
         startValue + (endValue - startValue) * this.progress;
     }
     if (this.progress === 1) {
+      for (let i = 0; i < this.propsList.length; i++) {
+        const propName = this.propsList[i] as keyof CoreNodeAnimatableProps;
+        const propValue = this.props[propName] as number;
+        this.node[propName] = propValue;
+      }
```
Comment (frank-weindel) [inline src/core/animations/CoreAnimation.ts:205]: Thanks! This should work fine but to squeeze maybe a tiny bit more performance out I think we could do a `progress === 1` check for line 197 and simply assign the `endValue` in that case.

Comment (marcel-danilewicz-consult-red) [reply src/core/animations/CoreAnimation.ts:205]: Do you mean something like this?
```
      if(this.progress===1){
        this.node[propName] = endValue;
      }
      else{
        this.node[propName] = startValue + (endValue - startValue) * this.progress;
      }
```
For some reason it doesn't work. Should I investigate further?

Comment (frank-weindel) [reply src/core/animations/CoreAnimation.ts:205]: Actually sorry I didn't even read the code close enough. You should be able to do it at the beginning of the for loop body:
```ts
if (this.progress === 1) {
  this.node[propName] = endValue;
  continue;
}
```

Comment (marcel-danilewicz-consult-red) [reply src/core/animations/CoreAnimation.ts]: Done

**Comment by frank-weindel** [comment]
@marcel-danilewicz-consult-red This issue has been fixed. Rebase onto dev

**Comment by marcel-danilewicz-consult-red** [comment]
It looks like this bug was also fixed in a commit, that is already present on dev, no need for rebasing and merging then.

**Comment by frank-weindel** [comment]
@marcel-danilewicz-consult-red Ah yeah it looks like @jfboeve addressed this as part of his changes for Shader Animations #314

**Pull request closed without merging**